### PR TITLE
Add support for env var override of gcp quota project

### PR DESCRIPTION
**Pull request by douglas-reid**
Sometimes, it is necessary to override the quota project for GCP (in addition to just the consumer project, as is done with GCP_METADATA). This PR provides a mechanism for that override.

[X] Policies and Telemetry

### CODE REVIEW

File: pkg/bootstrap/platform/gcp.go [hunk 1]
```diff
@@ -42,9 +42,11 @@ const (
 	GCEInstanceID        = "gcp_gce_instance_id"
 	GCEInstanceTemplate  = "gcp_gce_instance_template"
 	GCEInstanceCreatedBy = "gcp_gce_instance_created_by"
+	GCPQuotaProject      = "gcp_quota_project"
 )
 
 var GCPMetadata = env.RegisterStringVar("GCP_METADATA", "", "Pipe separted GCP metadata, schemed as PROJECT_ID|PROJECT_NUMBER|CLUSTER_NAME|CLUSTER_ZONE").Get()
+var gcpQuotaProjectVar = env.RegisterStringVar("GCP_QUOTA_PROJECT", "", "Allows specification of a quota project to be used in requests to GCP APIs.")
```
Comment (howardjohn) [inline pkg/bootstrap/platform/gcp.go:49]: does it make sense for GCPMetadata to allow an (optional) additional segment?

Comment (howardjohn) [reply pkg/bootstrap/platform/gcp.go:49]:
```suggestion
var GcpQuotaProjectVar = env.RegisterStringVar("GCP_QUOTA_PROJECT", "", "Allows specification of a quota project to be used in requests to GCP APIs.").Get()
```

For consistency and to allow override from other packages

File: pkg/bootstrap/platform/gcp_test.go [hunk 2]
```diff
@@ -258,6 +260,34 @@ func TestGCPMetadata(t *testing.T) {
...
+			meta := NewGCP().Metadata()
+			val, found := meta[GCPQuotaProject]
+			if got, want := found, v.wantFound; got != want {
+				tt.Errorf("Metadata() returned an unexpected value for GCPQuotaProject; found value: %t, want %t", got, want)
+			}
+			if got, want := val, v.wantProject; got != want {
+				tt.Errorf("Incorrect value for GCPQuotaProject; got = %q, want = %q", got, want)
+			}
+			os.Unsetenv("GCP_QUOTA_PROJECT")
+			shouldFillMetadata = metadata.OnGCE
```
Comment (qfel) [inline pkg/bootstrap/platform/gcp_test.go:286]: Maybe wrap in defer right after the corresponding set calls for extra maintainability

File: pkg/bootstrap/platform/gcp.go (hunk 1 above)
Comment (douglas-reid) [reply pkg/bootstrap/platform/gcp.go]: OK. Updated to allow for override.

File: pkg/bootstrap/platform/gcp_test.go (hunk 2 above)
Comment (douglas-reid) [reply pkg/bootstrap/platform/gcp_test.go]: sg. done.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**